### PR TITLE
Adjust catalog overview to allow for more text in the descriptions.

### PR DESCRIPTION
**Pull request by oponder**
We'd like to add some more text to the catalog descriptions, currently that doesn't work so good in the layout we have.

These are some quick changes for now.

Screenshot 2019-07-05 at 11 19 01 PM

### CODE REVIEW

**Comment by oponder** [comment]
I also noticed that the top level "Apps" isn't really totally correct. I'm ok with that in the navigation, but really the first level of this flow is the 'App Catalogs' so I made a quick change for that as well.

Screenshot 2019-07-05 at 10 39 11 PM

**Comment by marians** [comment]
With https://github.com/giantswarm/happa/pull/637 we now have `react-markdown`. What do you think about using that for displaying of the catalog description? With the descriptions as they are, it would result in the URL being linked. Plus it would allow to add formatting to descriptions, if this is what we want.

Regarding security, I think we should check whether we can completely ensure the descriptions are coming from trustworthy sources. Of not, react-markdown has options to disable embedded HTML, to prevent XSS.

**Comment by oponder** [comment]
> I think we should check whether we can completely ensure the descriptions are coming from trustworthy sources.

We're making the descriptions, it's not coming from the index.yaml of the app catalog repo or something like that. 

So I think it's ok. The description is a field on the App Catalog CR, which we create. Even the App Catalog CR's that point to partner or customer repo's will probably be created by us, or by the customer themselves.

Disabling embedded HTML regardless might be a nice preventative measure.

**Comment by oponder** [comment]
@marians ok, I added react-markdown! It looks to me like the default options are 'safe'. i.e. it disabled embedded html by default: https://github.com/rexxars/react-markdown#options

**Comment by marians** [comment]
Will test now on ginger

File: src/styles/components/_app_catalog.sass [hunk 1]
```diff
@@ -75,43 +75,49 @@
...
   p
     margin-bottom: 0px
-    padding: 10px 20px
     border-bottom-left-radius: 5px
     border-bottom-right-radius: 5px
+    font-size: 14px
 
   &:hover
     text-decoration: none
     box-shadow: 0px 0px 8px lighten(#1e4156, 22%);
```
Comment (marians) [inline src/styles/components/_app_catalog.sass]:
```suggestion
    box-shadow: 0px 0px 8px lighten(#1e4156, 22%)
```

Unrelated fix

Comment (marians) [reply src/styles/components/_app_catalog.sass]: Please double-check the white space here. Not sure whether github got this right.

File: src/components/app_catalog/catalogs/index.js [hunk 2]
```diff
@@ -30,24 +28,28 @@ class Catalogs extends React.Component {
...
+                          {this.props.catalogs.items[catalogName].spec.title}
+                        </h3>
+                        <p>
+                          <ReactMarkdown>
+                            {
+                              this.props.catalogs.items[catalogName].spec
+                                .description
+                            }
+                          </ReactMarkdown>
+                        </p>
```
Comment (marians) [inline src/components/app_catalog/catalogs/index.js]: This is resulting in nested `<p>` tags, as ReactMardown is wrapping every paragraph in `<p></p>`, too. You can remove the outer one.

**Comment by oponder** [comment]
Adjusted line length:

Screenshot 2019-07-08 at 5 15 29 PM

**Comment by marians** [comment]
FYI: I suggested `max-width` with unit `em`, as it is useful to let the line length depend on the font size. That's what the typographs from the old days taught me at least.

**Comment by oponder** [comment]
Would be useful if the font size was changing while it was resizing, or if we supported some kind of font-size change feature.

The way browsers zoom now is also not solely by increasing the font size, so I think we're ok.

But happy to rethink the typography and units usage (em / px) throughout happa at some point.
